### PR TITLE
Adding functionality for Agent Alerts

### PR DESCRIPTION
**Pull request by BrooksCunningham**
The intention for this PR is to add the ability to manage Agent Alerts via Terraform. 

https://docs.fastly.com/en/ngwaf/managing-agent-alerts

### CODE REVIEW

File: provider/resource_site_agent_alert.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+					}
+					return nil, []error{errors.New("interval must be 5, 10, or 60")}
+				},
+			},
+			"threshold": {
+				Type:        schema.TypeInt,
+				Description: "The number of occurrences of the tag in the interval needed to trigger the alert. Min 0, Max 10000",
+				Required:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if existsInRange(val.(int), 0, 10000) {
```
Comment (shawnps) [inline provider/resource_site_agent_alert.go:49]:
```suggestion
					if existsInRange(val.(int), 1, 10000) {
```

File: provider/resource_site_agent_alert.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if existsInInt(val.(int), 5, 10, 60) {
+						return nil, nil
+					}
+					return nil, []error{errors.New("interval must be 5, 10, or 60")}
+				},
+			},
+			"threshold": {
+				Type:        schema.TypeInt,
+				Description: "The number of occurrences of the tag in the interval needed to trigger the alert. Min 0, Max 10000",
```
Comment (shawnps) [inline provider/resource_site_agent_alert.go:46]:
```suggestion
				Description: "The number of occurrences of the tag in the interval needed to trigger the alert. Min 1, Max 10000",
```

File: provider/resource_site_agent_alert.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+				ForceNew:    true,
+			},
+			"tag_name": {
+				Type:        schema.TypeString,
+				Description: "The name of the tag whose occurrences the alert is watching. Must match an existing tag",
+				Required:    true,
+			},
+			"long_name": {
+				Type:        schema.TypeString,
+				Description: "description",
```
Comment (shawnps) [inline provider/resource_site_agent_alert.go:30]: Did you mean to leave this as `"description"`?

File: docs/resources/site_agent_alert.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `action` (String) Action for agent alert.
+- `interval` (Number) Integer value for interval. Must be 5, 10 or 60.
+- `site_short_name` (String) Site short name
+- `tag_name` (String) The name of the tag whose occurrences the alert is watching. Must match an existing tag
+- `threshold` (Number) The number of occurrences of the tag in the interval needed to trigger the alert. Min 0, Max 10000
```
Comment (shawnps) [inline docs/resources/site_agent_alert.md:24]:
```suggestion
- `threshold` (Number) The number of occurrences of the tag in the interval needed to trigger the alert. Min 1, Max 10000
```

File: provider/resource_site_agent_alert.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+			},
+			"threshold": {
+				Type:        schema.TypeInt,
+				Description: "The number of occurrences of the tag in the interval needed to trigger the alert. Min 0, Max 10000",
+				Required:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if existsInRange(val.(int), 0, 10000) {
+						return nil, nil
+					}
+					return nil, []error{errors.New("threshold must be between 1 and 10000")}
```
Comment (shawnps) [inline provider/resource_site_agent_alert.go:52]:
```suggestion
					return nil, []error{errors.New("threshold must be between 0 and 10000")}
```

**Comment by BrooksCunningham** [comment]
Thanks @shawnps for the suggestion. I have accepted your suggestion.

Is anything else needed before it may be merged with the main branch?

**Review by shawnps — APPROVED** [review]

**Pull request merged**